.github/workflows/enos-run.yml

```diff
@@ -2,157 +2,154 @@
+      VAULT_METADATA: ${{ inputs.vault-edition }}
+      # Pass in the matrix and matrix group for filtering
+      MATRIX_FILE: ./.github/enos-run-matrices/${{ inputs.matrix-file-name }}.json
+      MATRIX_TEST_GROUP: ${{ inputs.matrix-test-group }}
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Create Enos scenario matrix
-        id: enos-matrix
+      - uses: actions/checkout@v3
+      - id: metadata
         run: |
-          [[ ${{ env.ARTIFACT_NAME }} == *"ent"* ]] && scenarioFile=$(cat ./.github/enos-run-matrices/${{ env.ARTIFACT_SOURCE }}-ent.json |jq -c .) || scenarioFile=$(cat ./.github/enos-run-matrices/${{ env.ARTIFACT_SOURCE }}-oss.json |jq -c .)
-          echo "matrix=$scenarioFile" >> $GITHUB_OUTPUT
-  # Run Integration tests on Enos scenario matrix
-  enos:
-    name: Integration
-    needs: read-enos-matrix
+          echo "build-date=$(make ci-get-date)" >> $GITHUB_OUTPUT
+          echo "version=$(make ci-get-version)" >> $GITHUB_OUTPUT
+          filtered=$(make ci-filter-matrix)
+          echo "matrix=$(echo $filtered)}" >> $GITHUB_OUTPUT
+
+  # Run the Enos test scenarios
+  run:
+    needs: metadata
     strategy:
       fail-fast: false # don't fail as that can skip required cleanup steps for jobs
-      matrix: ${{ fromJson(needs.read-enos-matrix.outputs.enos-scenarios) }}
+      matrix: ${{ fromJson(needs.metadata.outputs.matrix) }}
     runs-on: ubuntu-latest
     env:
       GITHUB_TOKEN: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+      # Pass in enos variables
+      ENOS_VAR_aws_region: ${{ matrix.aws_region }}
+      ENOS_VAR_aws_ssh_keypair_name: ${{ inputs.ssh-key-name }}
+      ENOS_VAR_aws_ssh_private_key_path: ./support/private_key.pem
+      ENOS_VAR_tfc_api_token: ${{ secrets.TF_API_TOKEN }}
+      ENOS_VAR_artifactory_username: ${{ secrets.ARTIFACTORY_USER }}
+      ENOS_VAR_artifactory_token: ${{ secrets.ARTIFACTORY_TOKEN }}
+      ENOS_VAR_terraform_plugin_cache_dir: ./support/terraform-plugin-cache
+      ENOS_VAR_vault_build_date: ${{ needs.metadata.outputs.build-date }}
+      ENOS_VAR_vault_product_version: ${{ needs.metadata.outputs.version }}
+      ENOS_VAR_vault_revision: ${{ inputs.vault-revision }}
+      ENOS_VAR_vault_bundle_path: ./support/downloads/${{ inputs.build-artifact-name }}
+      ENOS_VAR_vault_license_path: ./support/vault.hclic
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Set up Terraform
-        uses: hashicorp/setup-terraform@v2
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
         with:
           # the Terraform wrapper will break Terraform execution in Enos because
           # it changes the output to text when we expect it to be JSON.
           terraform_wrapper: false
-      - name: Configure AWS credentials
-        uses: aws-actions/configure-aws-credentials@v1
+      - uses: aws-actions/configure-aws-credentials@v1-node16
```

Comment:
Curious why is the step name removed here ?

Reply:
I don't think they're useful as they literally say nearly the same thing when rendered.

Reply:
I'm also always trying to make these workflows shorter and simpler.

Reply:
I find the `name` makes it easier in the Actions UI to identify the step you want to view the details of